src/Parallel/Main.hpp

```diff
@@ -488,6 +507,42 @@ void Main<Metavariables>::pup(PUP::er& p) noexcept {  // NOLINT
   // options will be held in various code objects that will themselves be
   // serialized.
   p | phase_change_decision_data_;
+
+  p | checkpoint_dir_counter_;
+  // Check if next checkpoint dir is available
+  if constexpr (Algorithm_detail::has_WriteCheckpoint_v<
+                    typename Metavariables::Phase>) {
+    if (p.isUnpacking()) {
+      current_checkpoint_dir();  // to perform check if dir already exists
+    }
+  }
+
+  // For now we only support restarts on the same hardware configuration (same
+  // number of nodes and same procs per node) used when writing the checkpoint.
+  // We check this by adding counters to the pup stream.
+  if (p.isPacking()) {
```

Comment:
is the below `else` behavior really correct for the `isSizing` state? I'm confused on how that's not hitting the `ERROR`

Reply:
I hadn't thought of this, but now I'm confused too. I must be missing something about sizing puppers. But would you feel better if I inverted the logic to check `if (unpacking) { .. } else { ..}` ?

Reply:
yes, that would resolve my concerns fully :) I think in the sizing case you should pretty much always do the same thing you do in the packing case